sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1255,6 +1255,34 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     assert(testData.select($"*").showString(1, vertical = true) === expectedAnswer)
   }
 
+  test("SPARK-23023 Cast rows to strings in showString") {
+    val df1 = Seq(Seq(1, 2, 3, 4)).toDF("a")
+    assert(df1.showString(10) ===
```

Comment:
Do you know why it shows `WrappedArray` before?

Reply:
Since `RowEncoder` deserializer converts nested arrays into `WrappedArray`, `toString` shows do so in `showString`.

https://github.com/apache/spark/blob/55dbfbca37ce4c05f83180777ba3d4fe2d96a02e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/RowEncoder.scala#L304